workerapi/worker.thrift

```diff
@@ -1,37 +1,42 @@
 enum Status {
   UNKNOWN = 0
-  READY = 1
-  PENDING = 2
-  RUNNING = 3
-  COMPLETED = 4
+  PENDING = 1
+  RUNNING = 2
+  COMPLETE = 3
+  FAILED = 4
   ABORTED = 5
-  TIMEOUT = 6
-}
-
-//TODO: add useful load information when it comes time to have multiple runs.
-struct WorkerStatus {
-  1: required list<string> running
+  TIMEDOUT = 6
 }
 
 struct RunStatus {
   1: required Status status
-  2: optional string runId
+  2: required string runId
   3: optional string outUri
   4: optional string errUri
-  5: optional string info
+  5: optional string error
   6: optional i32 exitCode
 }
 
+// VersionId will be empty by default. To change this, pass a getVersion() func to server.Handler.
+// - Semantics are up to the implementer, this needn't be a library version, it could be a gitSha, etc.
+//
+// TODO: add useful load information when it comes time to have multiple runs.
+struct WorkerStatus {
+  1: required list<RunStatus> runs  # All runs excepting what's been Erase()'d
+  2: optional string versionId      # Version associated with this worker.
+}
+
 struct RunCommand {
-  1: required list<string> argv
-  2: optional string snapshotId
-  3: optional i32 timeoutMs
+  1: required list<string> argv       # Binary followed by any number of arguments.
+  2: optional map<string,string> env  # Binary followed by any number of arguments.
+  3: optional string snapshotId       # Scheme'd id, could be a patchId, sha1, etc.
+  4: optional i32 timeoutMs           # Kill the job if it hasn't completed in time (Status.TIMEOUT).
 }
 
 //TODO: add a method to kill the worker if we can articulate unrecoverable issues.
 service Worker {
-  WorkerStatus QueryWorker()
-  RunStatus Run(1: RunCommand cmd)
-  RunStatus Query(1: string runId)
-  RunStatus Abort(1: string runId)
+  WorkerStatus QueryWorker()         # Overall worker node status.
+  RunStatus Run(1: RunCommand cmd)   # Run a command and return job Status.(RUNNING|FAILED).
```

Comment:
nit: delete Running|Failed since there are more potential statuses that could be returned